### ISSUE TITLE
New 3DS XL White Screen on Launch

### ISSUE THREAD

**Issue by North-West-Wind** [DS-Homebrew/TWiLightMenu#1170]
**Your console configuration:** New 3DS XL with Luma3DS CFW 10.2 and FW 11.13.0-45U
**TWiLight Menu++ Version:** 16.1.0
**TWiLight Menu++ Theme:** I don't know because I cannot launch it.

### Expected behaviour
Get into the menu.

### Actual behaviour
My 3DS is stuck at the white screen.

### Steps to reproduce

1. Download `TWiLightMenu-3DS.7z` from v16.1.0, copy `_nds`, `roms` and `BOOT.NDS` to root, the 2 CIAs to my `/cias/` folder.
2. Install those CIAs.
3. Put my NDS roms into the `/roms/nds` folder.
4. Launch Twilight Menu++ and get stuck at the white screen.

### What I have tried so far

Reformat SD card (64GB) to FAT32 and 32KB allocated unit size.
Delete the `_nds` folder and Twilight Menu++ titles and install via Twilight Menu++ Updater.
Removing all the roms from the roms folder.
Run TWPatcher and make some widescreen thing.
Follow the guide at [3DS Hacks Guide](https://3ds.hacks.guide/troubleshooting#dsi--ds-functionality-is-broken-after-completing-the-guide)
Perform system update.
Update Twilight Menu++ via Updater.
Update nds-bootstrap via Updater.

**Comment by SleepyLark** [comment]
sometimes it takes a while to boot up the first time, have you tried waiting?  also have you tried seeing if other DS mode apps work like the Nintendo DS wifi connection page in the System Settings or DS Download Play?  If those also get white screens then it means your TWL_FIRM is broken (there's a way to fix it on 3ds.hacks.guide)

**Comment by North-West-Wind** [comment]
Yes I waited like 10 minutes. I did follow the guide to fix TWL_FIRM. DS Download Play works.

**Comment by abderrahmane-tj** [comment]
I've just had a similar issue with DSi XL. I noticed that it was booting up with the wifi led on.
I disabled the Wifi from Twilight Menu settings, now it loads quickly

**Comment by RocketRobz** [comment]
> I've just had a similar issue with DSi XL. I noticed that it was booting up with the wifi led on.
> I disabled the Wifi from Twilight Menu settings, now it loads quickly

That makes no sense.
All it does is just turn on/off WiFi, which takes no additional time on startup.

**Comment by abderrahmane-tj** [comment]
> > I've just had a similar issue with DSi XL. I noticed that it was booting up with the wifi led on.
> > I disabled the Wifi from Twilight Menu settings, now it loads quickly
> 
> That makes no sense.
> All it does is just turn on/off WiFi, which takes no additional time on startup.

I believe you 🤷🏼

**Comment by gek169** [comment]
This isn't the first time there's been a known issue, with a known fix, that rob couldn't explain.

I suspect there are a lot of things deep in the dependencies that cause issues like these, but are cropping up now either due to poor documentation or assumptions made in the development of the deps.

It'd be worthwhile if you could enable stacktraces @RocketRobz for twilight menu and nds-bootstrap.

If this already exists, maybe you could ask users to enable stack tracing and upload the trace along with their bug report when applicable?

I know there are logging options for Twilight menu, but they probably don't detail code being run in dependencies.

I'm thinking about how one might do stack tracing so as to be useful in non-crash situations (Obviously a stack trace is not going to have calls that have been popped off, so if it's only recorded once during the process, it's useless)

you might be able to get another thread (I don't know how well you can do multithreading on DSi hardware. It has multiple processors, but how well can you use them in tandem?) that periodically traps the other in a semaphore or barrier, then writes the other thread's call stack to file.

this would of course greatly slow down the code, but you could set it up so that it only runs if a certain button is held during Twilightmenu/NDS-bootstrap's startup, or a toggle in settings.ini. The stack trace logging would then occur every X milliseconds or clock cycles until the program shut off.

This might also help you discover bugs in No$GBA (Which I know you can do detailed stack tracing/logging in) because you'll be able to compare how code performs on real hardware as compared to the emulator (Many emulators actually emulate multiple processors with a single thread and simply rely on the fact that code must be written in a certain way to prevent race conditions. This can operate both to make inaccurate emulations and to hinder the development of software if the emulator is used for development testing. I don't know how accurately/well No$GBA emulates the multiple processors in DSes and GBAs. It's a thought.)

**Comment by fernandofleury** [comment]
Had the exact same issue, and turning off the wifi on the 3ds settings before booting twilight menu actually fixed it.

**Comment by xerohour** [comment]
I had this problem and disabling wifi on the 3ds allowed TWmenu to boot correctly. May not make sense but bugs usually do not.

**Comment by GrandLamia** [comment]
So, I turned off Wifi to get TWL-Menu to load--then went through the settings, and notice that the Auto-Region setting required HiyaCFW Firmware installed... which since I am running TWL-Menu on a 3DS I do not have HiyaCFW currently installed

I switched it over to USA Region, exited out of settings. Had it get stuck on the White Screen--so hit Home Button to exit. Tried reentering into TWL-Menu to see if I had rebroke it. It ran

Then as an experiment, I tried running it with WiFi turned on--because you know YOLO and what not... and now it works without disabling Wifi

This means, that you could potentially fix this error by editing the Conf/Ini Files by hand after mounting the MicroSD-Card on a computer... which while that works, possibly might not be a good way to do things

A possible fixes would be to have future releases on the 3DS not default to requiring a Firmware not common to the 3DS--or just change the default conf/ini setting to USA. The latter is a quick hack for a fix--and not particularly ideal

**Comment by Magicctoe** [comment]
disabling wifi did allow it to boot correctly.

**Comment by RocketRobz** [comment]
1. What happened before you disabled it?
2. How did you disable it?

**Comment by RocketRobz** [comment]
@Magicctoe Also, which version are you using?

**Comment by Sunriser45** [comment]
If anyone followed all the steps in this thread and TwilightMenu still doesn't load, check if your SD card (if you use one bigger than 32 gigs) is correctly formatted to FAT32 with NO empty partitions! It still wasn't working for me but doing that fixed it. You can't do it with Windows' default partitioner but a quick google search will find you tons of options!

**Comment by epic0722** [comment]
this doesnt work for me on my O3ds. i have correct partitions and and everything on my 3ds works and even ds wifi settings work and twmenu will not get past this white screen.

**Comment by FinTriesCode** [comment]
I have injected homebrew onto the SD card and followed instructions from there.

It runs .cias files ok

The issue is adding backwards compatibility (built in ds games). I have gotten as far as adding them to the device and having them show on the home screen however, when clicked on, it just loads a white screen and then gets suck (the white screen that usually progressed to the "nintendo ds" bootstrap/intro screen). sometimes i can go to home screen but nothing happens on reboot or relaunch.

Twilight Menu++ does the same thing.

I have all (that Im aware of) software required:
TwiLight Menu++
NDS Bootleg
NDS Forwarder

Still all white screen on boot.

I have tried a reformatting of the SD card, reinstall of all mentioned apps, readding new dsl files and purging files followed by another reinstall.

Been at it for hours lol
